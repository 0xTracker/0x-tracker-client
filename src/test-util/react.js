import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import React from 'react';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

export { renderWithRouter };
