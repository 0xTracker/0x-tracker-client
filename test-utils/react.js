import { createMemoryHistory } from 'history';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { Router } from 'react-router-dom';
import React from 'react';

import reducer from '../src/redux/reducer';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

const renderWithAppContext = (
  component,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState,
    store = createStore(reducer, initialState),
  } = {},
) => ({
  ...render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  ),
  history,
});

export { renderWithAppContext, renderWithRouter };
