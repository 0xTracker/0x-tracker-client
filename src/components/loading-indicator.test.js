import { act, render } from '@testing-library/react';
import React from 'react';

import LoadingIndicator from './loading-indicator';

beforeAll(() => {
  jest.useFakeTimers();
});

it('should render after 300ms', () => {
  const { container } = render(<LoadingIndicator />);

  act(() => {
    jest.advanceTimersByTime(299);
  });

  expect(container.firstChild).toBeNull();

  act(() => {
    jest.advanceTimersByTime(1);
  });

  expect(container.firstChild).not.toBeNull();
});

it('renders spinner by default', () => {
  const { container } = render(<LoadingIndicator />);

  act(() => {
    jest.runAllTimers();
  });

  expect(container.firstChild).toMatchSnapshot();
});

it('renders cylon', () => {
  const { container } = render(<LoadingIndicator type="cylon" />);

  act(() => {
    jest.runAllTimers();
  });

  expect(container.firstChild).toMatchSnapshot();
});

it('renders centered indicator', () => {
  const { container } = render(<LoadingIndicator centered />);

  act(() => {
    jest.runAllTimers();
  });

  expect(container.firstChild).toMatchSnapshot();
});
