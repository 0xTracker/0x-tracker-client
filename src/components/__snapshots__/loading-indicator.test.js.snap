// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders centered indicator 1`] = `
<div
  class="loading-indicator__AlignCenter-sc-1hkz55d-0 eiTrfJ"
>
  <div
    style="fill: currentColor; height: 50px; width: 50px;"
  >
    <svg
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
    >
      
  
      <path
        d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4"
        opacity=".25"
      />
      
  
      <path
        d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z"
      >
        
    
        <animatetransform
          attributeName="transform"
          dur="0.8s"
          from="0 16 16"
          repeatCount="indefinite"
          to="360 16 16"
          type="rotate"
        />
        
  
      </path>
      

    </svg>
    

  </div>
</div>
`;

exports[`renders cylon 1`] = `
<div
  style="fill: currentColor; height: 50px; width: 50px;"
>
  <svg
    viewBox="0 0 32 32"
    xmlns="http://www.w3.org/2000/svg"
  >
    
  
    <path
      d="M0 12 V20 H4 V12z"
      transform="translate(0 0)"
    >
      
    
      <animatetransform
        attributeName="transform"
        begin="0"
        calcMode="spline"
        dur="1.5s"
        keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
        keyTimes="0;0.3;0.6;1"
        repeatCount="indefinite"
        type="translate"
        values="0 0; 28 0; 0 0; 0 0"
      />
      
  
    </path>
    
  
    <path
      d="M0 12 V20 H4 V12z"
      opacity="0.5"
      transform="translate(0 0)"
    >
      
    
      <animatetransform
        attributeName="transform"
        begin="0.1s"
        calcMode="spline"
        dur="1.5s"
        keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
        keyTimes="0;0.3;0.6;1"
        repeatCount="indefinite"
        type="translate"
        values="0 0; 28 0; 0 0; 0 0"
      />
      
  
    </path>
    
  
    <path
      d="M0 12 V20 H4 V12z"
      opacity="0.25"
      transform="translate(0 0)"
    >
      
    
      <animatetransform
        attributeName="transform"
        begin="0.2s"
        calcMode="spline"
        dur="1.5s"
        keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
        keyTimes="0;0.3;0.6;1"
        repeatCount="indefinite"
        type="translate"
        values="0 0; 28 0; 0 0; 0 0"
      />
      
  
    </path>
    

  </svg>
  

</div>
`;

exports[`renders spinner by default 1`] = `
<div
  style="fill: currentColor; height: 50px; width: 50px;"
>
  <svg
    viewBox="0 0 32 32"
    xmlns="http://www.w3.org/2000/svg"
  >
    
  
    <path
      d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4"
      opacity=".25"
    />
    
  
    <path
      d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z"
    >
      
    
      <animatetransform
        attributeName="transform"
        dur="0.8s"
        from="0 16 16"
        repeatCount="indefinite"
        to="360 16 16"
        type="rotate"
      />
      
  
    </path>
    

  </svg>
  

</div>
`;
